projects/cardiovascular-patient-app/src/app/report-bp/custom-error-state-matcher.spec.ts

```diff
@@ -0,0 +1,29 @@
+import { FormControl, FormGroupDirective, NgForm, Validators } from '@angular/forms';
+import { CustomErrorStateMatcher } from './custom-error-state-matcher';
+
+describe('CustomErrorStateMatcher', () => {
+    let customErrorStateMatcher: CustomErrorStateMatcher;
+
+    beforeEach(() => {
+        customErrorStateMatcher = new CustomErrorStateMatcher();
+    });
+
+    it('should return false if form is untouched after being submitted', () => {
+        const control = new FormControl();
+        const form = {
+            submitted: true
+        } as FormGroupDirective | NgForm;
+
+        expect(customErrorStateMatcher.isErrorState(control, form)).toBe(false);
+    });
+
+    it('should return true if form is touched and invalid after being submitted', () => {
```

Comment:
This test is marking the control as dirty. The description should match what the test is doing.
```suggestion
    it('should return true if form is dirty and invalid after being submitted', () => {
```